### PR TITLE
Bug/fix part of pages assigned to dashboard

### PR DESCRIPTION
**Pull request by petar-qb**
## Description
If users create a Page object that contains action(s) without assigning it to the Dashboard.pages, browser console errors will appear.

The bug is currently reproducible in the main branch by configuring `default/app.py` in a following way (and by opening the app in the browser):
```
# Page object initialisation
create_country_analysis()

dashboard = vm.Dashboard(
    pages=[
        create_home_page(),
        create_variable_analysis(),
        create_relation_analysis(),
        create_continent_summary(),
        # create_country_analysis(),
    ],
    navigation=vm.Navigation(
        pages={
            "Analysis": ["Homepage", "Variable Analysis", "Relationship Analysis", "Country Analysis"],
            "Summary": ["Continent Summary"],
        }
    ),
)
```

---

### Testing

The unit test that catches this issue will be implemented once we write tests for the ActionLoop.

@l0uden Should we add this check into integration tests? 

---

## Screenshot
image

image

## Checklist

- [x] I have not referenced individuals, products or companies in any commits, directly or indirectly
- [x] I have not added data or restricted code in any commits, directly or indirectly
- [x] I have updated the docstring of any public function/class/model changed
- [x] I have added the PR number to the change description in the changelog fragment, e.g. `Enable feature XXX (#123)` (if applicable)

## Types of changes

- [ ] Docs/refactoring (non-breaking change which improves codebase)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: vizro-core/src/vizro/actions/_action_loop/_action_loop_utils.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+"""Contains utilities to create the extract Action and ActionsChain models only from registered pages."""
```
Comment (huong-li-nguyen) [inline vizro-core/src/vizro/actions/_action_loop/_action_loop_utils.py:1]:
```suggestion
"""Contains utilities to extract the Action and ActionsChain models from registered pages only."""
```

File: vizro-core/src/vizro/actions/_action_loop/_action_loop_utils.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+def _get_actions_chains_on_registered_pages() -> List[ActionsChain]:
+    """Gets list of ActionsChain models for registered pages."""
+    actions_chains: List[ActionsChain] = []
+    for page_registry_obj in page_registry.values():
+        try:
+            page_model: Page = model_manager[page_registry_obj["module"]]  # type: ignore[assignment]
+        except KeyError:
+            continue
+        actions_chains.extend(_get_all_actions_chains_on_page(page=page_model))
```
Comment (huong-li-nguyen) [inline vizro-core/src/vizro/actions/_action_loop/_action_loop_utils.py:44]:
```suggestion
    for registered_page in page_registry.values():
        try:
            page: Page = model_manager[registred_page["module"]]  # type: ignore[assignment]
        except KeyError:
            continue
        actions_chains.extend(_get_all_actions_chains_on_page(page=page))
```

**Review by huong-li-nguyen — DISMISSED** [review]
Minor comments on naming and description, but the rest looks great 👍  Thanks for fixing!

**Review by huong-li-nguyen — APPROVED** [review]

**Review by nadijagraca — APPROVED** [review]
Looks good. Thanks for fixing this Peter.

**Comment by l0uden** [comment]
@petar-qb , added the check to the integration tests

**Pull request merged**